frontend/packages/ceph-storage-plugin/integration-tests/tests/2-tests/test-expand.scenario.ts

```diff
@@ -0,0 +1,76 @@
+import { browser, ExpectedConditions as until } from 'protractor';
+import { execSync } from 'child_process';
+import { clickKebabAction } from '@console/internal-integration-tests/views/crud.view';
+import { click } from '@console/shared/src/test-utils/utils';
+import { testName } from '@console/internal-integration-tests/protractor.conf';
+import { testPVC, testDeployment } from '../../mocks/expand-test-mocks';
+import { requestSizeValue } from '../../views/add-capacity.view';
+import {
+  goToPersistentVolumeClaims,
+  expandButton,
+  createNewPersistentVolumeClaim,
+  expandSizeOption,
+  capacityDropdown,
+} from '../../views/pvc.view';
+import { SIZE_UNITS } from '../../utils/consts';
+
+const expandValue = String(Number(testPVC.size) + 1);
+
+const createDeployment = () =>
+  execSync(`echo '${JSON.stringify(testDeployment)}' | kubectl create -f -`);
+
+describe('Tests Expand flow for PVC', () => {
+  beforeAll(async () => {
+    // Create a PVC
+    await createNewPersistentVolumeClaim(testPVC, true, createDeployment);
+    // Wait for it to bind
+    await goToPersistentVolumeClaims();
+  });
+
+  it('Test PVC can be expanded (In MiBs)', async () => {
+    await clickKebabAction(testPVC.name, 'Expand PVC');
+    await browser.wait(until.visibilityOf(requestSizeValue));
+    await requestSizeValue.sendKeys(expandValue);
+    await click(capacityDropdown);
+    await click(expandSizeOption(SIZE_UNITS.MI));
+    await click(expandButton);
+
+    const pvcJSON = JSON.parse(
+      execSync(`kubectl get pvc ${testPVC.name} -n ${testName} -o json`).toString(),
+    );
+    const requestedStorage = pvcJSON.spec.resources.requests.storage;
+    expect(requestedStorage.trim()).toEqual(`${expandValue}${SIZE_UNITS.MI}`);
+  });
+
+  it('Test PVC can be expanded (In GiBs)', async () => {
+    await goToPersistentVolumeClaims();
+    await clickKebabAction(testPVC.name, 'Expand PVC');
+    await browser.wait(until.visibilityOf(requestSizeValue));
+    await requestSizeValue.sendKeys(expandValue);
+    await click(capacityDropdown);
+    await click(expandSizeOption(SIZE_UNITS.GI));
+    await click(expandButton);
```

Comment:
Since the codes inside each `it` block is checking the same functionality, better approach would be to create a common function and pass the unit to it.

Reply:
yes, it looks like the logic can indeed be factored out.